tests/test_callprofiler.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2024, Quentin Tsai <quentin.tsai.tw@gmail.com>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+import time
+import re
+import os
+from functools import wraps
+import modmesh
+
+
+# https://docs.python.org/3/library/functools.html#functools.wraps
+def profile_function(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        modmesh.call_profiler.start_caller(func.__name__)
+        result = func(*args, **kwargs)
+        modmesh.call_profiler.end_caller()
+        return result
+    return wrapper
+
+
+class CallProfilerTC(unittest.TestCase):
+
+    def test_singleton(self):
+        self.assertIs(modmesh.call_profiler, modmesh.CallProfiler.instance)
+
+    def test_reset(self):
+
+        @profile_function
+        def foo1():
+            time.sleep(3)
+
+        foo1()
+        modmesh.call_profiler.reset()
+        result = modmesh.call_profiler.print_profiling_result()
+        self.assertIn("Profiling Result", result)
+        self.assertNotIn("foo1", result)
+
+    @unittest.skipUnless("nt" != os.name,
+                         "timing code on windows has resoltion issue")
```

Comment:
Notice there's time resolution issue in windows, causing the CI to fail as the measured time is always less than what it should be.
Wonder if there's a better way to handle this on windows.

Reply:
Much much less it is in windows than in other systems?

Preferably this can be fixed (or worked around) in this PR.  We use the `skipUnless` trick if can't do it soon enough.

Reply:
can we have a workaround for Windows? e.g. increasing the time or enlarging the error tolerance?